base/deprecated.jl

```diff
@@ -838,16 +838,16 @@ end
 @deprecate big(r::StepRange) big.(r)
 @deprecate big(r::StepRangeLen) big.(r)
 @deprecate big(r::LinSpace) big.(r)
-@deprecate big{T<:Integer,N}(x::AbstractArray{T,N}) big.(x)
-@deprecate big{T<:AbstractFloat,N}(x::AbstractArray{T,N}) big.(x)
+@deprecate big(x::AbstractArray{<:Integer}) big.(x)
+@deprecate big(x::AbstractArray{<:AbstractFloat}) big.(x)
 @deprecate big(A::LowerTriangular) big.(A)
 @deprecate big(A::UpperTriangular) big.(A)
 @deprecate big(A::Base.LinAlg.UnitLowerTriangular) big.(A)
 @deprecate big(A::Base.LinAlg.UnitUpperTriangular) big.(A)
 @deprecate big(B::Bidiagonal) big.(B)
-@deprecate big{T<:Integer,N}(A::AbstractArray{Complex{T},N}) big.(A)
-@deprecate big{T<:AbstractFloat,N}(A::AbstractArray{Complex{T},N}) big.(A)
-@deprecate big{T<:Integer,N}(x::AbstractArray{Complex{Rational{T}},N}) big.(A)
+@deprecate big(A::AbstractArray{<:Complex{<:Integer}}) big.(A)
+@deprecate big(A::AbstractArray{<:Complex{<:AbstractFloat}}) big.(A)
+@deprecate big(x::AbstractArray{<:Complex{<:Rational{<:Integer}}}) big.(A)
```

Comment:
`AbstractArray{Complex{<:Integer}}` is equivalent to `AbstractArray{Complex{T} where T<:Integer}` so it matches `AbstratctArray`s with a `UnionAll` element type (`Complex{T} where T<:Integer`).

```julia
foo(::AbstractArray{Complex{<:Integer}}) = 1
bar(::AbstractArray{<:Complex{<:Integer}}) = 2
foo([1im]) #MethodError
bar([1im]) == 2
foo((Complex{T} where T<:Integer)[1im]) == 1
bar((Complex{T} where T<:Integer)[1im]) == 2
# since AbstractArray{Complex{T} where T<:Integer} <: (AbstractArray{T} where T<:(Complex{I} where I))
```